cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs

```diff
@@ -1345,22 +1347,50 @@ pProposalFiles sbe balExUnits =
 
 pProposalFile
   :: BalanceTxExecUnits
-  -> Parser (ProposalFile In, Maybe (ScriptWitnessFiles WitCtxStake))
+  -> Parser (ProposalFile In, Maybe CliProposalScriptRequirements)
 pProposalFile balExUnits =
   (,)
     <$> pFileInDirection "proposal-file" "Filepath of the proposal."
     <*> optional (pProposingScriptOrReferenceScriptWitness balExUnits)
  where
   pProposingScriptOrReferenceScriptWitness
-    :: BalanceTxExecUnits -> Parser (ScriptWitnessFiles WitCtxStake)
+    :: BalanceTxExecUnits -> Parser CliProposalScriptRequirements
   pProposingScriptOrReferenceScriptWitness bExUnits =
-    pScriptWitnessFiles
-      WitCtxStake
+    pProposalScriptWitness
       bExUnits
       "proposal"
       Nothing
       "a proposal"
-      <|> pPlutusStakeReferenceScriptWitnessFilesVotingProposing "proposal-" balExUnits
+      <|> pProposalReferencePlutusScriptWitness "proposal-" balExUnits
```

Comment:
I would add the `-` in the inner function for consistency with `pProposalScriptWitness`. It also makes more sense to have the connectors in the function that makes the connection because it is only one place.